nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/position/LegacyBlobStorageEventPositionProvider.java

```diff
@@ -0,0 +1,154 @@
+        final Map<String, EventPosition> partitionEventPosition;
+
+        if (containerExists()) {
+            final BlobListDetails blobListDetails = new BlobListDetails().setRetrieveMetadata(true);
+            final ListBlobsOptions listBlobsOptions = new ListBlobsOptions().setPrefix(consumerGroup).setDetails(blobListDetails);
+            final Iterable<BlobItem> blobItems = blobContainerAsyncClient.listBlobs(listBlobsOptions).toIterable();
+            partitionEventPosition = getPartitionEventPosition(blobItems);
+        } else {
+            partitionEventPosition = Collections.emptyMap();
+        }
+
+        return partitionEventPosition;
+    }
+
+    private Map<String, EventPosition> getPartitionEventPosition(final Iterable<BlobItem> blobItems) {
+        final Map<String, EventPosition> partitionEventPosition = new LinkedHashMap<>();
+
+        for (final BlobItem blobItem : blobItems) {
+            if (Boolean.TRUE.equals(blobItem.isPrefix())) {
+                continue;
+            }
+
+            final String partitionId = getPartitionId(blobItem);
+            final EventPosition eventPosition = getEventPosition(blobItem);
+            if (eventPosition == null) {
+                logger.info("Legacy Event Position not found for Partition [{}] Blob [{}]", partitionId, blobItem.getName());
+            } else {
+                partitionEventPosition.put(partitionId, eventPosition);
+            }
+        }
+
+        return partitionEventPosition;
+    }
+
+    private String getPartitionId(final BlobItem blobItem) {
+        final String blobItemName = blobItem.getName();
+        final Path blobItemPath = Paths.get(blobItemName);
+        final Path blobItemFileName = blobItemPath.getFileName();
+        return blobItemFileName.toString();
+    }
+
+    private EventPosition getEventPosition(final BlobItem blobItem) {
+        final EventPosition eventPosition;
+
+        final String blobName = blobItem.getName();
+        final BlobAsyncClient blobAsyncClient = blobContainerAsyncClient.getBlobAsyncClient(blobName);
+
+        if (itemExists(blobAsyncClient)) {
+            final BinaryData content = blobAsyncClient.downloadContent().block();
+            if (content == null) {
+                throw new IllegalStateException(String.format("Legacy Event Position content not found [%s]", blobName));
+            }
+
+            try {
+                // Read com.microsoft.azure.eventprocessorhost.AzureBlobLease from JSON
+                final JsonNode lease = objectMapper.readTree(content.toBytes());
+                if (lease.hasNonNull(LEASE_SEQUENCE_NUMBER_FIELD)) {
+                    final JsonNode sequenceNumberField = lease.get(LEASE_SEQUENCE_NUMBER_FIELD);
+                    final long sequenceNumber = sequenceNumberField.asLong();
+                    eventPosition = EventPosition.fromSequenceNumber(sequenceNumber);
+
+                    blobAsyncClient.delete().block();
```

Comment:
Thanks for highlighting this issue @turcsanyip.

Part of the challenge with reading the legacy offset information is that the consumer identifiers change between SDK versions. The identifiers are based on a UUID, and the new SDK generates a new UUID.

The other issue is that the Processor needs a way to avoid using the legacy offset information after initial retrieval, as all subsequent handling should be based on the new checkpoint store strategy.  The most straightforward way is to delete the legacy offset information. In this scenario, the first instance of the Processor should read the legacy offset information and proceed. Other instances should begin reading from the latest offset and start tracking using the new checkpoint store strategy. With this approach, at least one consumer will retrieve events based on legacy offset information. It should not be necessary for all consumers to start from previous offsets as long as at least one consumer reads the events from the oldest offset. That might require a small adjustment to this legacy provider.

This is probably something worth noting in migration guidance, but open to additional ideas.